features/FEATURE_BLE/ble/Gap.h

```diff
@@ -321,6 +269,617 @@ class GapAdvertisingData;
+         */
+        uint8_t size;
+
+        /**
+         * Capacity of the array holding the addresses.
+         */
+        uint8_t capacity;
+    };
+
+
+    /** Check controller support for a specific feature.
+     *
+     * @param feature Feature to check.
+     * @return True if feature is supported.
+     */
+    virtual bool isFeatureSupported(ble::controller_supported_features_t feature)
+    {
+        // TODO: deal with legacy implementation
+    }
+
+    /*                                     advertising                                           */
+
+    /** Return currently available number of supported advertising sets.
+     *  This may change at runtime.
+     *
+     * @return Number of advertising sets that may be created at the same time.
+     */
+    virtual uint8_t getMaxAdvertisingSetNumber();
+
+    /** Return maximum advertising data length supported.
+     *
+     * @return Maximum advertising data length supported.
+     */
+    virtual uint8_t getMaxAdvertisingDataLength();
+
+    /** Create an advertising set and apply the passed in parameters. The handle returned
+     *  by this function must be used for all other calls that accept an advertising handle.
+     *  When done with advertising, remove from the system using destroyAdvertisingSet().
+     *
+     * @note The exception is the ble::LEGACY_ADVERTISING_HANDLE which may be used at any time.
+     *
+     * @param[out] handle Advertising handle returned, valid only if function returned success.
+     * @param parameters Advertising parameters for the newly created set.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t createAdvertisingSet(
+        ble::advertising_handle_t *handle,
+        const ble::AdvertisingParameters &parameters
+    );
+
+    /** Remove the advertising set (resets its set parameters). The advertising set must not
+     *  be active.
+     *
+     * @note ble::LEGACY_ADVERTISING_HANDLE may not be destroyed.
+     *
+     * @param handle Advertising set handle.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t destroyAdvertisingSet(ble::advertising_handle_t handle);
+
+    /** Set advertising parameters of an existing set.
+     *
+     * @param handle Advertising set handle.
+     * @param params New advertising parameters.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t setAdvertisingParameters(
+        ble::advertising_handle_t handle,
+        const ble::AdvertisingParameters &params
+    );
+
+    /** Set new advertising payload for a given advertising set.
+     *
+     * @param handle Advertising set handle.
+     * @param payload Advertising payload.
+     * @param minimiseFragmentation Preference for fragmentation.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t setAdvertisingPayload(
+        ble::advertising_handle_t handle,
+        mbed::Span<uint8_t> payload,
+        bool minimiseFragmentation = false
```

Comment:
Might be worth explaining why you'd ever set `minimiseFragmentation` to `false` (why would I prefer fragmentation ;)?). Also it should be `minimizeFragmentation` as we're supposed to write in US-English :)

Reply:
I fixed it with a bazooka; I removed the parameter as it is not very useful for the user right now.
It may be reintroduced latter if it is necessary.